File: chall00/juligonz.c
x/* ************************************************************************** */
/*                                                                            */
/*                                                        :::      ::::::::   */
/*   juligonz.c                                         :+:      :+:    :+:   */
/*                                                    +:+ +:+         +:+     */
/*   By: juligonz <juligonz@student.42.fr>          +#+  +:+       +#+        */
/*                                                +#+#+#+#+#+   +#+           */
/*   Created: 2020/04/06 15:17:15 by juligonz          #+#    #+#             */
/*   Updated: 2020/04/06 15:37:31 by juligonz         ###   ########.fr       */
/*                                                                            */
/* ************************************************************************** */

#include <string.h>
#include <stdio.h>

int ft_necklace(char *s1, char *s2)
{
	size_t i;
	size_t j;
	size_t k;

	if (strlen(s1) != strlen(s2))
		return (0);
	j = 0;
	while (s2[j])
	{
		if (s1[0] == s2[j])
		{
			i = 0;
			k = j;
			while (s1[i] && s1[i] == s2[k])
			{
				k = (!s2[k + 1] ? 0 : k + 1);
				i++;
			}
			if (!s1[i])
				return (1);
		}
		j++;
	}
	return (0);
}
/*
int main(int ac, char **av)
{
	(void)ac;
	printf("%d (\"%s\", \"%s\")\n", 
		   ft_necklace(av[1], av[2]),
					   av[1], av[2]);
	return (0);
}
*/

File: chall02/juligonz.py
#!/usr/bin/env python3

# **************************************************************************** #
#                                                                              #
#                                                         :::      ::::::::    #
#    juligonz.py                                        :+:      :+:    :+:    #
#                                                     +:+ +:+         +:+      #
#    By: juligonz <juligonz@student.42.fr>          +#+  +:+       +#+         #
#                                                 +#+#+#+#+#+   +#+            #
#    Created: 2020/04/20 14:15:41 by juligonz          #+#    #+#              #
#    Updated: 2020/04/20 15:28:46 by juligonz         ###   ########.fr        #
#                                                                              #
# **************************************************************************** #

import sys

dict_morse = {
    'a' : ".-",
    'b' : "-...",
    'c' : "-.-.",
    'd' : "-..",
    'e' : ".",
    'f' : "..-.",
    'g' : "--.",
    'h' : "....",
    'i' : "..",
    'j' : ".---",
    'k' : "-.-",
    'l' : ".-..",
    'm' : "--",
    'n' : "-.",
    'o' : "---",
    'p' : ".--.",
    'q' : "--.-",
    'r' : ".-.",
    's' : "...",
    't' : "-",
    'u' : "..-",
    'v' : "...-",
    'w' : ".--",
    'x' : "-..-",
    'y' : "-.--",
    'z' : "--..",
    ' ' : " "
}

def print_usage():
    print("usage:", sys.argv[0], "<a-zA-Z string>")

if len(sys.argv) != 2 or not sys.argv[1]:
    print_usage()
    sys.exit()

try:
    print("".join([dict_morse[c]  for c in sys.argv[1].lower()]))
except KeyError:
    print_usage()
